src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -544,31 +565,93 @@ private Boolean isPassable(final BlockState blockState)
+            if (path != null) {
+                // Unstuck with path
+                ArrayList<BlockPos> checkPositions = new ArrayList<>();
+                PathPoint next = path.getPathPointFromIndex(path.getCurrentPathIndex());
+                // Blocks in front of the worker
+                checkPositions.add(new BlockPos(next.x, next.y, next.z));
+                checkPositions.add(new BlockPos(next.x, next.y + 1, next.z));
+                // Block above the worker
+                checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY() + 2, worker.getCurrentPosition().getZ()));
+
+                mineIfEqualsBlockTag(checkPositions, BlockTags.LEAVES);
+                return;
+            }
+        }
+
+        // General unstuck
+        ArrayList<BlockPos> checkPositions = new ArrayList<>();
+        // Blocks in front of the worker
+        checkPositions.add(new BlockPos(worker.getForward().getX(), worker.getForward().getY(), worker.getForward().getZ()));
+        checkPositions.add(new BlockPos(worker.getForward().getX(), worker.getForward().getY() + 1, worker.getForward().getZ()));
+
+        // blocks in north of the player
+        checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY(), worker.getCurrentPosition().getZ()).north());
+        checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY() + 1, worker.getCurrentPosition().getZ()).north());
+
+        // blocks in east of the player
+        checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY(), worker.getCurrentPosition().getZ()).east());
+        checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY() + 1, worker.getCurrentPosition().getZ()).east());
+
+        // blocks in south of the player
+        checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY(), worker.getCurrentPosition().getZ()).south());
+        checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY() + 1, worker.getCurrentPosition().getZ()).south());
+
+        // blocks in west of the player
+        checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY(), worker.getCurrentPosition().getZ()).west());
+        checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY() + 1, worker.getCurrentPosition().getZ()).west());
+
+        mineIfEqualsBlockTag(checkPositions, BlockTags.LEAVES);
+    }
+
+    /**
+     * Checks blocks for tag and mines the first it fines if its the same
+     * @param blockPositions block positions
+     * @param tag tag to check
+     */
+    private boolean mineIfEqualsBlockTag(List<BlockPos> blockPositions, Tag<Block> tag){
```

Comment:
This function isnt going to work nicely like this, because our mineBlock() is made to mine one position after another as it works through repeated calls and a delay inbetween

Reply:
Why? 
Its only adding one block per call

Reply:
ye one block per call isnt enough, you need to call the same position twice, with a time delay inbetween. It simulates the harvesting with a delay before actually breaking the block. Since the AI repeatedly executes this code. You should do if(leaf) mine(leaf) and then you do nothing but wait for the next time it runs into it again, and you check for the first leaf you find again and break that one again.

Reply:
But that was happening before, I just moved it into a separate function to minimize the code lines

Reply:
no before it was an if else, it did check first pos -> is leaf? -> mine block. Next AI update it does check first pos -> is leaf? -> mine block (block breaks). Then next AI update it does check first pos -> no leaf -> check second pos -> is leaf -> mine second

Reply:
And now its the same
I starts with the first object in the list
if leaf -> mine and return, second run, again first is leaf -> mine(break) -> return
Then first is not leaf -> check second, if leaf ...